packages/medical-domain/src/domains/repositories/statistics/glycemia-statistics.service.ts

```diff
@@ -0,0 +1,278 @@
+      result.total += duration
+      return result
+    },
+    {
+      veryLow: 0,
+      low: 0,
+      target: 0,
+      high: 0,
+      veryHigh: 0,
+      total: 0
+    }
+  )
+
+  if (numDays > 1) {
+    return {
+      veryLow: durationInRange.veryLow / durationInRange.total * MS_IN_DAY,
+      low: durationInRange.low / durationInRange.total * MS_IN_DAY,
+      target: durationInRange.target / durationInRange.total * MS_IN_DAY,
+      high: durationInRange.high / durationInRange.total * MS_IN_DAY,
+      veryHigh: durationInRange.veryHigh / durationInRange.total * MS_IN_DAY,
+      total: durationInRange.total
+    }
+  }
+  return durationInRange
+}
+
+function getReadingsInRangeData(smbgData: Smbg[], bgBounds: BgBounds, numDays: number, dateFilter: DateFilter): CbgRangeStatistics {
+  const filteredSmbg = SmbgService.filterOnDate(smbgData, dateFilter.start, dateFilter.end, getWeekDaysFilter(dateFilter))
+  const readingsInRange = filteredSmbg.reduce(
+    (result, smbg) => {
+      const classification = classifyBgValue(bgBounds, smbg.value, ClassificationType.FiveWay)
+      result[classification]++
+      result.total++
+      return result
+    },
+    {
+      veryLow: 0,
+      low: 0,
+      target: 0,
+      high: 0,
+      veryHigh: 0,
+      total: 0
+    }
+  )
+  if (numDays > 1) {
+    return {
+      veryLow: readingsInRange.veryLow / numDays,
+      low: readingsInRange.low / numDays,
+      target: readingsInRange.target / numDays,
+      high: readingsInRange.high / numDays,
+      veryHigh: readingsInRange.veryHigh / numDays,
+      total: readingsInRange.total
+    }
+  }
+  return readingsInRange
+}
+
+function getSensorUsage(cbgData: Cbg[], numDays: number, dateFilter: DateFilter): SensorUsageStatistics {
+  const filteredCbg = CbgService.filterOnDate(cbgData, dateFilter.start, dateFilter.end, getWeekDaysFilter(dateFilter))
+  const totalDuration = getCgmTotalDuration(filteredCbg)
+
+  const total = Math.round(numDays * MS_IN_DAY)
+
+  return {
+    sensorUsage: totalDuration,
+    total
+  }
+}
+
+function getAverageGlucoseData(bgData: Cbg[] | Smbg[], dateFilter: DateFilter): AverageGlucoseStatistics {
+  const filteredBg = CbgService.filterOnDate(bgData as Cbg[], dateFilter.start, dateFilter.end, getWeekDaysFilter(dateFilter)) as Bg[]
+  const totalBgValue = filteredBg.reduce((sum, bgData) => sum + bgData.value, 0)
+  const numberOfBg = filteredBg.length
+  return {
+    averageGlucose: totalBgValue / numberOfBg,
+    total: numberOfBg
+  }
+}
+
+function getCoefficientOfVariationData(bgData: Cbg[] | Smbg[], dateFilter: DateFilter): CoefficientOfVariationStatistics {
+  const filteredBg = CbgService.filterOnDate(bgData as Cbg[], dateFilter.start, dateFilter.end, getWeekDaysFilter(dateFilter))
+  const average = getAverageGlucoseData(filteredBg, dateFilter)
+  const noDataStatistics = {
+    insufficientData: true,
+    coefficientOfVariation: Number.NaN,
+    total: average.total
+  }
+  if (average.total < 3) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏What does this means? What does `3` represents here? `average.total` looks weird to me! How can an average have a total?

Reply:
The total is the number of Cbg used to process this average... So I admit the name is weird, but I just revamped legacy interfaces.

So 3 is just the minimum number of cbg we must have in order to process this statistics.

Reply:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏Thanks. Can we have a const with such as `const MIN_CBG_FOR_VARIATION_COEF=3`? I think that it could improve this a bit